scipy/signal/_filter_design.py

```diff
@@ -1428,100 +1464,133 @@ def zpk2sos(z, p, k, pairing='nearest'):
     # 4. Further optimizations of the section ordering / pole-zero pairing.
     # See the wiki for other potential issues.
 
-    valid_pairings = ['nearest', 'keep_odd']
+    if pairing is None:
+        pairing = 'minimal' if analog else 'nearest'
+
+    valid_pairings = ['nearest', 'keep_odd', 'minimal']
     if pairing not in valid_pairings:
         raise ValueError('pairing must be one of %s, not %s'
                          % (valid_pairings, pairing))
-    if len(z) == len(p) == 0:
-        return array([[k, 0., 0., 1., 0., 0.]])
 
-    # ensure we have the same number of poles and zeros, and make copies
-    p = np.concatenate((p, np.zeros(max(len(z) - len(p), 0))))
-    z = np.concatenate((z, np.zeros(max(len(p) - len(z), 0))))
-    n_sections = (max(len(p), len(z)) + 1) // 2
-    sos = zeros((n_sections, 6))
+    if analog and pairing != 'minimal':
+        raise ValueError('for analog zpk2sos conversion, '
+                         + 'pairing must be "minimal"')
```

Comment:
```suggestion
        raise ValueError('for analog zpk2sos conversion, '
                         'pairing must be "minimal"')
```

Reply:
can't say I'm a fan of implicit string concatenation, but ok.

Reply:
It's not pythonic to concatenate strings like that. And redundant in this case.